### ISSUE TITLE
Error:java: ElementAnnotationUtil.getLocationTypeADT: invalid location TYPE_ARGUMENT(0) for type: Object

### ISSUE THREAD

**Issue by justmin** [typetools/checker-framework#989]
Simple interfaces compilation produce exception with CF 2.1.6. Works good with CF 2.1.0

ListWrapper.java:

```
import org.checkerframework.checker.nullness.qual.NonNull;
import org.checkerframework.checker.nullness.qual.Nullable;

import java.util.List;

public interface ListWrapper<E> extends List<@NonNull E> {
    default @Nullable E distinct() {
        if(size() > 1) throw new RuntimeException();
        return isEmpty() ? null : get(0);
    }
}

```

Test.java
```
import org.checkerframework.checker.nullness.qual.Nullable;

public interface Test {

    ListWrapper<Boolean> getWrapper();

    default @Nullable Boolean getBoolean(){
        return getWrapper().distinct();
    }
}
```

**Comment by smillst** [comment]
Thanks for reporting this issue.  I added a simplified testcase, checker/tests/nullness/Issue989.java

**Comment by justmin** [comment]
Is this issue will be fixed? It's a showstopper for me preventing migration to 2.1.6+

**Comment by wmdietl** [comment]
Sorry about the delay. I'll try to get to it soon.

**Comment by wmdietl** [comment]
I think this was caused by the fix for #880, which undid #876.
PR #1253 attempts to fix both #880 and #989. Let's see whether downstream Daikon tests fail.
